### PR TITLE
XMLSerializer: Correct Map value serialization

### PR DESCRIPTION
**Pull request by stanio**
Found while working on [CORE-3549][]...

Given test:

```java
import liquibase.change.custom.CustomChangeWrapper;
import liquibase.precondition.CustomPreconditionWrapper;
import liquibase.serializer.core.xml.XMLChangeLogSerializer;

        CustomPreconditionWrapper customPrecondition = new CustomPreconditionWrapper();
        customPrecondition.setClassName("liquibase.precondition.ExampleCustomPrecondition");
        customPrecondition.setParam("name", "test_1");
        customPrecondition.setParam("count", "31");

        CustomChangeWrapper customChange = new CustomChangeWrapper();
        customChange.setClassLoader(Thread.currentThread().getContextClassLoader());
        customChange.setClass("liquibase.change.custom.ExampleCustomSqlChange");
        customChange.setParam("tableName", "tab_name");
        customChange.setParam("columnName", "col_name");
        customChange.setParam("newValue", "");
        
        XMLChangeLogSerializer xmlSerializer = new XMLChangeLogSerializer();
        System.out.println(xmlSerializer.serialize(customPrecondition, true));
        System.out.println(xmlSerializer.serialize(customChange, true));
```

Actual result:

```xml
<customPrecondition className="liquibase.precondition.ExampleCustomPrecondition" params="name">
    <paramValues name="test_1"/>
    <paramValues count="31"/>
</customPrecondition>
<customChange class="liquibase.change.custom.ExampleCustomSqlChange">
    <param>
        <newValue/>
    </param>
    <param>
        <tableName>tab_name</tableName>
    </param>
    <param>
        <columnName>col_name</columnName>
    </param>
</customChange>
```

Expected result:

```xml
<customPrecondition className="liquibase.precondition.ExampleCustomPrecondition">
    <param name="name" value="test_1"/>
    <param name="count" value="31"/>
</customPrecondition>
<customChange class="liquibase.change.custom.ExampleCustomSqlChange">
    <param name="tableName" value="tab_name"/>
    <param name="columnName" value="col_name"/>
    <param name="newValue" value=""/>
</customChange>
```

[CORE-3549]: https://liquibase.jira.com/browse/CORE-3549 (Implement "compileChangeLogXml" command)

┆Issue is synchronized with this [Jira Story](https://datical.atlassian.net/browse/LB-17) by [Unito](https://www.unito.io/learn-more)
┆Fix Versions: Liquibase 3.10.1

### CODE REVIEW

File: liquibase-core/src/main/java/liquibase/serializer/core/xml/XMLChangeLogSerializer.java [hunk 1]
```diff
@@ -197,7 +197,8 @@ private void setValueOnNode(Element node, String objectNamespace, String objectN
         } else if (value instanceof Map) {
             for (Map.Entry entry : (Set<Map.Entry>) ((Map) value).entrySet()) {
                 Element mapNode = currentChangeLogFileDOM.createElementNS(LiquibaseSerializable.STANDARD_CHANGELOG_NAMESPACE, qualifyName(objectName, objectNamespace, parentNamespace));
-                setValueOnNode(mapNode, objectNamespace, (String) entry.getKey(), entry.getValue(), serializationType, objectNamespace);
+                setValueOnNode(mapNode, objectNamespace, "name", entry.getKey(), serializationType, objectNamespace);
+                setValueOnNode(mapNode, objectNamespace, "value", entry.getValue(), serializationType, objectNamespace);
```
Comment (stanio) [inline liquibase-core/src/main/java/liquibase/serializer/core/xml/XMLChangeLogSerializer.java]: There's some funny namespace handling in there (originally), I'm not entirely sure 1) it is all correct and 2) my change produces expected results in all cases.  The change works as expected for the `customPrecondition` and `customChange` at least, which are the only uses I've found so far.

**Comment by stanio** [comment]
The new revision tries to keep the results/behavior in a number of largely undefined(?) cases.

**Comment by stanio** [comment]
I've been given a reference on [writing unit tests for Liquibase Core](https://liquibase.jira.com/wiki/spaces/LB/pages/1274937609/How+to+Write+Liquibase+Core+Unit+Tests) in another PR.  I'll further have a look at adding a test for this.

**Comment by ro-rah** [comment]
Hey @stanio !

Good seeing another PR from you!  Just steadily working my way through our PRs!  Wanna give writing Unit Tests for this submission too?  As mentioned above we may not have hit all the cases and be nice if we could provide some coverage here:
liquibase-core/src/test/java/liquibase/precondition/CustomPreconditionWrapperTest.java 

Thanks!

Ronak

**Comment by stanio** [comment]
What licensing issue it has found in 6c711ee14cab4fafe403dc3b33ad2216e56f13b9 is out of my mind.

**Comment by ro-rah** [comment]
Hey @stanio , thanks, I'm moving this to conditioning.
Sized it as risk low:
- It has a test
- this issue was uncovered as a bug while working on https://liquibase.jira.com/browse/CORE-3549

**Pull request merged**